scribble-lib/scribble/private/run-pdflatex.rkt

```diff
@@ -2,16 +2,21 @@
 
 (require scheme/system scheme/port)
 
-(provide run-pdflatex run-dvipdf-latex)
+(provide run-pdflatex run-dvipdf-latex run-xelatex)
 
-(define (run-pdflatex file [notify void]) (run file notify #f))
+(define (run-pdflatex file [notify void]) (run file notify 'pdflatex))
 (define (run-dvipdf-latex file [notify void]) 
   (parameterize ([function-name 'run-dvipdf-latex])
-    (run file notify #t)))
+    (run file notify 'dvipdf)))
+(define (run-xelatex file [notify void])
+  (parameterize ([function-name 'run-xelatex])
+    (run file notify 'xelatex)))
 
 (define max-runs 5)
-(define (run file notify via-dvipdf?)
-  (define latex-cmd-name (if via-dvipdf? "latex" "pdflatex"))
+(define (run file notify type)
+  (define latex-cmd-name (cond [(equal? type 'pdflatex) "pdflatex"]
```

Comment:
minor point: since these are symbols you can use `eq?` or `(case type [(pdflatex) "pdflatex"] ....)`